### PR TITLE
feat(meta): support group notification

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Support group notification (tables, indexes, sinks, sources ...).
- DDL statements like adding columns and dropping columns could cause catalog inconsistency at frontend nodes, so we need group notification to provide atomicity and consistency.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/src/scheduler/plan_fragmenter.rs [hunk 1]
```diff
@@ -947,7 +947,7 @@ impl BatchPlanFragmenter {
                 .get_fragment_mapping(&table_catalog.fragment_id)
                 .ok_or_else(|| {
                     anyhow!(
-                        "failed to get the vnode mapping for the `Materialize` of {}",
+                        "failed to get the vnode mapping for the `BatchLookupJoin` of {}",
```
Comment (BugenZhao) [inline src/frontend/src/scheduler/plan_fragmenter.rs:950]: I guess we were referring to the vnode mapping of `Materialize` fragment of the base table here. 🤔

File: proto/meta.proto [hunk 2]
```diff
@@ -280,6 +289,7 @@ message SubscribeResponse {
     backup_service.MetaBackupManifestId meta_backup_manifest_id = 17;
     SystemParams system_params = 19;
     hummock.WriteLimits hummock_write_limits = 20;
+    RelationGroup relation_group = 21;
```
Comment (BugenZhao) [inline proto/meta.proto:292]: I guess we can ignore the compatibility here if we're not going to support progressive upgrades in a cluster. 😁

**Review by BugenZhao — APPROVED** [review]
LGTM

**Review by yezizp2012 — APPROVED** [review]
LGTM

**Pull request merged**